stock-db/stock_db_solution.py

```diff
@@ -85,17 +89,63 @@
 msft_max_min_list = msft_max_min_query.fetchall()
 msft_max_min_list # max_date is '2014-07-21' and min_date is '2000-01-03'
 
-msft_max_min_df = pd.DataFrame(data = msft_max_min_list)
+# Obtaining column names of msft_table
+msft_table_cols = pd.read_sql("""
+                            SELECT *
+                            FROM msft_table
+                            ORDER BY Date;
+                            """, conn)
+msft_table_cols.columns
+
+msft_max_min_df = pd.DataFrame(data = msft_max_min_list, columns = ['max_date', 'min_date'])
 msft_max_min_df
 
-msft_max_min_table_query = cur.execute("""
+# Use ORDER BY for minimum date
+msft_min_table_query = cur.execute("""
                             SELECT *
                             FROM msft_table
-                            WHERE Date = '2014-07-21' OR Date = '2000-01-03';
+                            WHERE Date IS NOT NULL
+                            ORDER BY DATE ASC
+                            LIMIT 1;
                             """)
-msft_max_min_table_list = msft_max_min_table_query.fetchall()
-msft_max_min_table_df = pd.DataFrame(data = msft_max_min_table_list)
-msft_max_min_table_df
+msft_min_table_list = msft_min_table_query.fetchall()
+msft_min_table_df = pd.DataFrame(data = msft_min_table_list, columns = msft_table_cols.columns)
+msft_min_table_df
+
+# = (SELECT MAX(DATE) FROM msft_table) #
+# WHERE DATE = (MIN MAX STATEMETN)
+
+# Use ORDER BY for maximum date
+msft_max_table_query = cur.execute("""
+                            SELECT *
+                            FROM msft_table
+                            WHERE Date IS NOT NULL
+                            ORDER BY DATE DESC
+                            LIMIT 1;
+                            """)
+msft_max_table_list = msft_max_table_query.fetchall()
+msft_max_table_df = pd.DataFrame(data = msft_max_table_list, columns = msft_table_cols.columns)
```

Comment:
The instructions weren't totally clear on this but the max and min entries for each table (msft and aapl) should be contained in the same dataframe (i.e. two entries per dataframe). Having a single entry dataframe would be overkill.